test/test_groupby.jl

```diff
@@ -239,7 +239,7 @@ show(io,"text/plain",gv)
 
 if VERSION > v"1.7"
     # type inference does not work in julia 1.6
-    @test eltype(gv) == Array{Float32,3}
+    @test_broken eltype(gv) == Array{Float32,3}
```

Comment:
This PR apparently breaks this type inference. We should probably solve this.

Reply:
Have you had a look with Cthulhu.jl? just `@descend` into the type unstable call and look around at the types. You can paste the bad bits here if needed. Profiling can also help show the instability.

Reply:
This is fixed in: https://github.com/JuliaIO/DiskArrays.jl/pull/267